### PR TITLE
fix(rpc-password-generator): update password validation to match KDF password policy

### PR DESCRIPTION
**Pull request by takenagain**
The password generator could intermittently generate passwords with 3 repeated characters, which would be rejected by KDF according to the password policy.

## Summary by CodeRabbit

- **New Features**
  - Introduced robust password validation with detailed error feedback for requirements such as length, digit, letter case, special characters, forbidden substrings, and repeated characters.
  - Enhanced password generation to ensure all security requirements are met before returning a password.

- **Bug Fixes**
  - Passwords are now guaranteed to include all required character types and avoid forbidden patterns.

- **Tests**
  - Added comprehensive tests covering password validation and generation, including Unicode and edge cases.

### CODE REVIEW

**Comment by takenagain** [comment]
@coderabbitai review

File: packages/komodo_defi_types/lib/src/utils/security_utils.dart [hunk 1]
```diff
@@ -1,70 +1,143 @@
...
 
-  final allCharacters = upperCaseLetters +
-      lowerCaseLetters +
-      digits +
-      specialCharacters +
-      (extendedSpecialCharacters ? extendedSpecial : '');
+    // Fill the rest of the password length with random characters from the pool
+    for (var i = 4; i < length; i++) {
+      password.add(allCharacters[random.nextInt(allCharacters.length)]);
+    }
```
[the remark this answers is not in the document]
Comment (takenagain) [reply packages/komodo_defi_types/lib/src/utils/security_utils.dart]: Done in [b793ea1](https://github.com/KomodoPlatform/komodo-defi-sdk-flutter/pull/58/commits/b793ea11adbf2b6cbcf0866ec3aaa946caf4dd4e)
